docs/howto/smart_contract_upload.md

```diff
@@ -0,0 +1,138 @@
+    required in order to use and configue smart contracts. `admin_keys`
+    identify who is allowed to add permissions to a contract and `peer_services`
+    is for identifying which parties are needed for consensus.
+    
+
+    b. Vote to accept the circuit on node beta 
+    
+    ```
+    $ splinter circuit vote \
+        --key <path_to_beta_private_key> \
+        --url http://splinterd-beta:8085 my-circuit --accept
+    ```
+
+    You have now established a circuit between node alpha and node beta based
+    on the provided circuit definition
+
+    c. Verify the circuit has been established 
+
+    ```
+    $ splinter circuit list --url http://splinterd-alpha:8085
+
+    ID            MANAGEMENT MEMBERS
+    <circuit_id>  tutorial   alpha-node-000;beta-node-000
+    ```
+
+2. Package smart contract
+
+    a. Create manifest.yaml
+        
+    The manifest contains the inputs, outputs, version, and the contract
+    name. Below is a sample manifest
+
+    ```
+    name: my_contract
+    version: '1.0'
+    inputs:
+      - '5b7349'
+    outputs:
+      - '5b7349'
+    ```
+
+    b. Bundle the manifest and the compiled wasm for the smart contract into a
+       tarball that has the scar extension 
+
+    ```
+    $ tar -jcvf my_contract.scar my_contract.wasm manifest.yaml 
+    ```
+
+3. Create contract registry for smart contract
+
+    Each contract needs a contract registry to be created that keeps track of
+    versions of the smart contract as well as the contract's owners.
+
+    ```
+    $ scabbard cr create my_contract \
+        --owner <alpha_node_public_key> \
+        --key <path_to_alpha_node_private_key> \
+        --url http://splinterd-alpha:8085 \
+        --service-id my-circuit::scabbard-service-a
+    ```
+
+4. Upload the smart contract.
+
+   ```
+   $ scabbard contract upload ./my_contract.scar \
+       --key <path_to_alpha_node_private_key> \
+       --url http://splinterd-alpha:8085 \
+       --service-id my-circuit::scabbard-service-a
+   ```
+
+5. Create the namespace registry for the smart contract.
+
+    The namespace registry stores the namespace, the owners of the namespace,
+    and the permissions given to that namespace. A namespace is a state address
+    prefix used to identify a portion of state.
+
+   ```
+   $ scabbard ns create <namespace> \
+       --owner <alpha_node_public_key> \
+       --key <path_to_alpha_node_private_key> \
+       --url http://splinterd-alpha:8085 \
+       --service-id my-circuit::scabbard-service-a
+   ```
+
+6. Create contract permissions.
+
+    The `perm` command creates permissions that allow the smart contract to
+    access state. The below command gives the smart contract both read and
+    write permissions.
+
+   ```
+   $ scabbard perm <namespace> my_contract --read --write \
+       --key <path_to_alpha_node_private_key> \
+       --url http://splinterd-alpha:8085 \
+       --service-id my-circuit::scabbard-service-a
+   ```
+
```

Comment:
Can you add a step here for listing/showing the contract like in the Grid on Splinter example?